### PR TITLE
Bump version to v0.1.0+20241204.0.0

### PR DESCRIPTION
**Pull request by karencfv**
As per the last product meeting, we're moving away from beta. We'll now use the versioning format [oxide.rs uses](https://github.com/oxidecomputer/oxide.rs/releases)

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v0.1.0-beta10
+v0.1.0+20241204.0.0
```
Comment (sudomateo) [inline VERSION:1]: Should this be just one `.0` at the end like `oxide.rs`?

```suggestion
v0.1.0+20241204.0
```

**Review by sudomateo — COMMENTED** [review]
Asked a minor clarifying question around the version itself before approving.

File: VERSION (hunk 1 above)
Comment (karencfv) [reply VERSION:1]: Ah! So this is a recent change that came from dropshot https://github.com/oxidecomputer/omicron/pull/7050/files#diff-8e355f4f647bc8c52d26845412f351c2d246246825442453e135d69c4346c681R10

The version should also change in oxide.rs for this release AFAIK

Comment (sudomateo) [reply VERSION:1]: Thank you for the context!

**Review by sudomateo — APPROVED** [review]

**Pull request merged**